vision_agent/agent/vision_agent_v2.py

```diff
@@ -106,19 +141,21 @@ def write_and_exec_code(
     user_req: str,
     subtask: str,
     orig_code: str,
-    code_writer_call: Callable,
+    code_writer_call: Callable[..., str],
     model: LLM,
     tool_info: str,
     exec: Execute,
+    retrieved_ltm: str,
     max_retry: int = 3,
```

Comment:
NIT> Do you want to keep it to 2 ? consistent with previous agent because 3 might be too many iterations.

Reply:
With GPT-4o it's pretty fast now, I don't see too much of a reason to decrease it.